osf_tests/test_registration_moderation_notifications.py

```diff
@@ -0,0 +1,359 @@
+import datetime
+
+from osf.management.commands.add_notification_subscription import add_reviews_notification_setting
+from osf.migrations import update_provider_auth_groups
+from osf.models import NotificationDigest
+from osf.models.action import RegistrationAction
+from website.profile.utils import get_profile_image_url
+
+from osf_tests.factories import (
+    RegistrationFactory,
+    AuthUserFactory
+)
+
+from website import mails, settings
+
+from osf.utils.workflows import RegistrationModerationTriggers, RegistrationModerationStates
+
+from osf.utils.notifications import (
+    notify_submit,
+    notify_accept_reject,
+    notify_moderator_registration_requests_withdrawal,
+    notify_reject_withdraw_request
+)
+
+@pytest.mark.django_db
+class TestRegistrationMachineNotification:
+
+    MOCK_NOW = datetime.datetime(2018, 2, 4)
+
+    @pytest.yield_fixture(autouse=True)
+    def time_machine(self):
+        with mock.patch('osf.utils.machines.timezone.now', return_value=self.MOCK_NOW):
+            yield
+
+    @pytest.fixture()
+    def contrib(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def admin(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def registration(self, admin, contrib):
+        registration = RegistrationFactory(creator=admin)
+        registration.add_contributor(admin, 'admin')
+        registration.add_contributor(contrib, 'write')
+        update_provider_auth_groups()
+        return registration
+
+    @pytest.fixture()
+    def provider(self, registration):
+        return registration.provider
+
+    @pytest.fixture()
+    def moderator(self, provider):
+        user = AuthUserFactory()
+        provider.add_to_group(user, 'moderator')
+        return user
+
+    @pytest.fixture()
+    def accept_action(self, registration, admin):
+        registration_action = RegistrationAction.objects.create(
+            creator=admin,
+            target=registration,
+            trigger=RegistrationModerationTriggers.ACCEPT_SUBMISSION.value,
+            from_state=RegistrationModerationStates.INITIAL.value,
+            to_state=RegistrationModerationStates.ACCEPTED.value,
+            comment='yo'
+        )
+        return registration_action
+
+    @pytest.fixture()
+    def withdraw_request_action(self, registration, admin):
+        registration_action = RegistrationAction.objects.create(
+            creator=admin,
+            target=registration,
+            trigger=RegistrationModerationTriggers.REQUEST_WITHDRAWAL.value,
+            from_state=RegistrationModerationStates.ACCEPTED.value,
+            to_state=RegistrationModerationStates.PENDING_WITHDRAW.value,
+            comment='yo'
+        )
+        return registration_action
+
+    def test_submit_notifications(self, registration, moderator, admin, contrib, provider):
+        """
+        "As moderator of branded registry, I receive email notification upon admin author(s) submission approval"
```

Comment:
These are great! They would be even better if they included the REQS ticket number.